core/controllers/editor.py

```diff
@@ -701,33 +702,20 @@ def get(self, exploration_id):
                 'Unrecognized output format %s' % output_format)
 
 
-class StateDownloadHandler(EditorHandler):
-    """Downloads a state as a YAML string."""
+class StateYamlHandler(EditorHandler):
+    """Given a representation of a state, converts it to a YAML string."""
 
-    def get(self, exploration_id):
+    def get(self):
         """Handles GET requests."""
-        try:
-            exploration = exp_services.get_exploration_by_id(exploration_id)
-        except:
-            raise self.PageNotFoundException
+        state_dict = json.loads(self.request.get('stringified_state'))
```

Comment:
actually, why is it a better idea to pass the entire state into the url instead of what we were doing before?

It just seems really convoluted, the frontend has a state which it converts to json, and then passes to the backend, and then the backend checks if it actually is a state, and then passes the state back to the frontend?